tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -0,0 +1,330 @@
+    The URL format is http://geodesy.unr.edu/gps_timeseries/trop/<ssss>/<ssss>.<yyyy>.trop.zip
+    Inputs:
+        statID   - 4-character station identifier
+        year     - 4-numeral year
+    '''
+    URL = "{0}gps_timeseries/trop/{1}/{1}.{2}.trop.zip".format(baseURL, statID.upper(), year)
+    if download:
+        saveLoc = os.path.abspath(os.path.join(writeDir, '{0}.{1}.trop.zip'.format(statID.upper(), year)))
+        filepath = download_url(URL, saveLoc, verbose=verbose)
+    else:
+        filepath = check_url(URL, verbose=verbose)
+    return filepath
+
+
+def getStatsByllh(llhBox = None, baseURL = _UNR_URL, userstatList = None):
+    '''
+    Function to pull lat, lon, height, beginning date, end date, and number of solutions for stations inside the bounding box llhBox. 
+    llhBox should be a tuple with format (lat1, lat2, lon1, lon2), where lat1, lon1 define the lower left-hand corner and lat2, lon2 
+    define the upper right corner. 
+    '''
+    import pandas as pd
+    from urllib.request import urlopen 
+
+    if llhBox is None:
+        llhBox = [-90, 90, 0, 360]
+
+    stationHoldings = '{}NGLStationPages/DataHoldings.txt'.format(baseURL)
+    data = urlopen(stationHoldings) # it's a file like object and works just like a file
+    stations = []
+    for ind, line in enumerate(data): # files are iterable
+        if ind == 0:
+            continue
+        statID, lat, lon = getID(line)
+        # Only pass if in bbox
+        # And if user list of stations specified, only pass info for stations within list
+        if inBox(lat, lon, llhBox) and (not userstatList or statID in userstatList):
+            #convert lon into range [-180,180]
+            lon = fix_lons(lon)
+            stations.append({'ID': statID, 'Lat': lat, 'Lon': lon})
+    
+    print('{} stations were found'.format(len(stations)))
+    stations = pd.DataFrame(stations)
+    # Report stations from user's list that do not cover bbox
+    if userstatList:
+        userstatList=[i for i in userstatList if i not in stations['ID'].to_list()]
+        if userstatList:
+            print("Warning: The following user-input stations are not covered by the input bounding box {0}: {1}" \
+                .format(str(llhBox).strip('[]'),str(userstatList).strip('[]')))
+        
+    return stations
+
+
+def check_url(url, verbose = False):
+    '''
+    Check whether a file exists at a URL. Modified from 
+    https://stackoverflow.com/questions/9419162/download-returned-zip-file-from-url
+    '''
+    r = requests.get(url, stream=True)
+    if r.status_code==404:
+        return ''
+    else:
+        return url
+
+
+def download_url(url, save_path, verbose = False, chunk_size=128):
+    '''
+    Download a file from a URL. Modified from 
+    https://stackoverflow.com/questions/9419162/download-returned-zip-file-from-url
+    '''
+    r = requests.get(url, stream=True)
+    if r.status_code==404:
+        return ''
+    else:
```

Comment:
True, but since I want to return empty fields in case the URL isn't valid (see my above comment) I suppose I could set the url variable to an empty string. I.e.:
``
    if r.status_code == 404:
        url = ''
    return url
``

Reply:
If you simply delete the `else:` (And dedent the contents of the Else block) the function will behave exactly the same way. It’s a nice way of saving a level of indentation, which in turn makes it easier to read.